lms/djangoapps/grades/subsection_grade_factory.py

```diff
@@ -104,6 +106,13 @@ def update(self, subsection, only_if_higher=None, score_deleted=False, force_upd
             )
             self._update_saved_subsection_grade(subsection.location, grade_model)
 
+            COURSE_ASSESSMENT_GRADE_CHANGED.send_robust(
+                sender=None,
+                user=self.student,
+                subsection_id=text_type(calculated_grade.location),
```

Comment:
Nit: we're fully on python 3 now, you can just use the `str()` built-in, rather than `text_type`.

Reply:
Ooh, good point. I was just trying to match our functionality in the gradebook view (because we use the `text type` subsection id [here](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/grades/rest_api/v1/gradebook_views.py#L473). These need to match for us to link edx courses to the integrated courses on the external LMS'.

Reply:
and this also brings up the fact that we're using `text type` frequently in other places, like the receiver as well as the gradepoint view.... thinking about it now, we are taking the text type of this `text type'd` parameter in the receiver, so maybe we just pass the `calculated_grade.location` through unaltered?

Reply:
So as long the receiver converts it to string before we forward it to the async task we should be good, correct?